File: asgn3/questionsa.cpp
/** ---------------------------------------------
*
* @file		questionsa.cpp
* @author	Nick McCollum
* @version	1.0
*
* CS-202 - Assignment 3 - Abstract Classes
*		   Question class implementation
*
* -------------------------------------------- */

#include <iostream>
#include <algorithm>
#include "questionsa.h"

using namespace std;

// ----------------------------------------------
QuestionSA::QuestionSA(string questionText = "", string answerText = "")
	: Question(questionText, answerText)
{
	this->questionText = questionText;
	this->answerText = answerText;
	correct = false;
}

// ----------------------------------------------
void QuestionSA::showQuestion() {
	cout << questionText;
	cout << "?";
}

// ----------------------------------------------
bool QuestionSA::checkAnswer(string givenAnswer) {
	string actualAnswer = answerText;

	// Remove whitespace and convert to lower
	removeWhitespace(actualAnswer);
	removeWhitespace(givenAnswer);
	toLowerCase(actualAnswer);
	toLowerCase(givenAnswer);

	if (givenAnswer.compare(actualAnswer) != 0) {
		return false;
	}

	markCorrect();
	return true;
}

File: Quiz/QuestionSA.java
/**
 * CS-202 - Assignment 5 - Java Classes
 * Short Answer question object used by Quiz.
 * Displays SA question and checks the user's answer.
 * 
 * @author Nick McCollum
 * @version 1.0
 *
 */

public class QuestionSA extends Question {

	public QuestionSA(String question, String answer) {
		super(question, answer);
		this.question = question;
		this.answer = answer;
		correct = false;
	}

	// ------------------------------------------------------------------
	/**
	 * Displays the MC question text in the console.
	 */
	public void showQuestion() {
		System.out.print(question + "?");
	}
	
	// ------------------------------------------------------------------
	/**
	 * Checks if the answer provided by the user was correct.
	 * @param givenAnswer	The answer entered by the user.
	 * @return boolean	True if the answer was correct, false if not.
	 */
	public boolean checkAnswer(String givenAnswer) {
		givenAnswer = removeWhitespace(givenAnswer.toLowerCase());
		String actualAnswer = removeWhitespace(answer.toLowerCase());
		
		if (!givenAnswer.contentEquals(actualAnswer)) {
			return false;
		}
		
		markCorrect();
		return true;
	}
}
